### DISCUSSION TITLE
Quarkus' Dev Services/Live Reload while doing Keycloak extensions development

### DISCUSSION THREAD

**Discussion by larsw** [keycloak/keycloak#14351] [Archive]
Hi,

After the migration to Quarkus, is there a way of using Quarkus' Dev Services with Live Reload when doing development of custom Keycloak extensions?

If anyone has a good write-up on a good dev workflow including both live/low friction reload, it would be highly appreciated.

**Comment by schuerg** [comment]
This is currently not possible with the Quarkus Keycloak deployment. You need a keycloak server (or at least keycloak cluster node) restart for reloading providers/extensions.

> While with the WildFly distribution you could hot-deploy custom providers, change the database vendor as a runtime configuration this is no longer supported.  
> *Source: https://www.keycloak.org/migration/migrating-to-quarkus*

**Reply by fredericoschardong** [reply]
Is this still the case?

**Reply by Walnussbaer** [reply]
Does this mean that if I deploy a custom theme as a jar, there is no way around a server restart if I want to be able to activate that theme?

**Reply by shawkins** [reply]
Try running in quarkus dev mode: https://github.com/keycloak/keycloak/blob/main/quarkus/CONTRIBUTING.md#running-the-server-in-development-mode

I believe that changes to existing provider jars are picked up, but new jars aren't yet handled - I haven't tested this in a while though, so it may work now.

**Reply by Walnussbaer** [reply]
> Try running in quarkus dev mode: https://github.com/keycloak/keycloak/blob/main/quarkus/CONTRIBUTING.md#running-the-server-in-development-mode
> 
> I believe that changes to existing provider jars are picked up, but new jars aren't yet handled - I haven't tested this in a while though, so it may work now.

Thanks for the reply. I don't know whether changes to an existing jar are picked up on the fly, but hot load of new jars does NOT work, even if dev mode is enabled.

**Reply by pendula95** [reply]
This feature would be appreciated
